src/tensorflow.cc

```diff
@@ -1290,17 +1291,42 @@ AutoCompleteHelper::FixBatchingSupport()
   // agree. We need to update the configuration itself as well as the
   // cached value we have already initialized in the model state.
   if (max_batch_size == 0) {
-    const int new_max_batch_size = model_support_batching_ ? 1 : 0;
+    // 0 is a valid parameter for the default-max-batch-size but it is
+    // not valid for models which explicitly support batching.
+    const int new_max_batch_size =
+        model_support_batching_
+            ? std::max(
+                  model_state_->BackendConfig()->default_max_batch_size_, 1)
+            : 0;
 
     triton::common::TritonJson::Value mbs_value;
     model_state_->ModelConfig().Find("max_batch_size", &mbs_value);
     mbs_value.SetInt(new_max_batch_size);
 
+    // Turn on dynamic batching if no batching support is supplied by the
```

Comment:
Why not enable dynamic_batching when the user has specified `max_batch_size=10` in their model config.pbtxt?

Reply:
Will update to check this after setting the batch size.

Reply:
Moved this logic past max_batch_size setting.